### ISSUE TITLE
[azappconfig] GetSettings with cancelled context panics

### ISSUE THREAD

**Issue by pravinpushkar** [Azure/azure-sdk-for-go#19223]
### Bug Report

Below is my sample code - 

```
cContext, cancel := context.WithCancel(ctx)
cancel()
resp, err := r.client.GetSetting(
                   cContext,
                   key,
                  &azappconfig.GetSettingOptions{
	               Label: r.getLabelFromMetadata(req.Metadata),
                  },
)
```

The above code when takes cancelled context, it panics. What should be recommended behaviour here. Is this something that should be handled at the sdk level ?

Below is the part of the stacktrace - 

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x38 pc=0x1026fe94c]

goroutine 124 [running]:
github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig.(*syncTokenPolicy).Do(0x1400000e698, 0x14001196700)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/data/azappconfig@v0.4.1/policy_sync_token.go:99 +0x1ac
github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/exported.(*Request).Next(0x140011966c0)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/internal/exported/request.go:84 +0xf8
github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig.(*hmacAuthenticationPolicy).Do(0x14001042a68?, 0x140011966c0)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/data/azappconfig@v0.4.1/policy_hmac_auth.go:75 +0x5ec
github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/exported.(*Request).Next(0x14001196680)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/internal/exported/request.go:84 +0xf8
github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime.(*retryPolicy).Do(0x14001042d08?, 0x14001196680)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/runtime/policy_retry.go:125 +0x51c
github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/exported.(*Request).Next(0x14001196640)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/internal/exported/request.go:84 +0xf8
github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime.telemetryPolicy.Do({{0x140001247c0?, 0x104c9bb00?}}, 0x14001196640)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/runtime/policy_telemetry.go:66 +0x188
github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/exported.(*Request).Next(0x14001196600)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/internal/exported/request.go:84 +0xf8
github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime.includeResponsePolicy(0x14001196600)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/runtime/policy_include_response.go:19 +0x20
github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime.policyFunc.Do(0x14001042eb8?, 0x101ce26d4?)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/runtime/pipeline.go:72 +0x28
github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/exported.(*Request).Next(0x140011965c0)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/internal/exported/request.go:84 +0xf8
github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/exported.Pipeline.Do({{0x1400012cc00?, 0x10512fc78?, 0x14001196540?}}, 0x140011965c0)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azcore@v1.1.3/internal/exported/pipeline.go:96 +0x160
github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig/internal/generated.(*AzureAppConfigurationClient).GetKeyValue(0x1400067b200, {0x10512fc78?, 0x14001196540?}, {0x14001190563?, 0x0?}, 0x0?)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/data/azappconfig@v0.4.1/internal/generated/azureappconfiguration_client.go:477 +0x6c
github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig.(*Client).GetSetting(0x14000677170, {0x10512fc78, 0x14001196540}, {0x14001190563, 0x7}, 0x14000679ce0)
	/Users/pravin/go_projects/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/data/azappconfig@v0.4.1/client.go:293 +0x13c
github.com/dapr/components-contrib/configuration/azure/appconfig.(*ConfigurationStore).Get(0x14000a39830, {0x105133168, 0x1400119a000}, 0x140008d1fa0)
	/Users/pravin/go_projects/src/github.com/dapr/components-contrib/configuration/azure/appconfig/appconfig.go:198 +0x1ec

Last line is the sample code above.

**Comment by pravinpushkar** [comment]
@antkmsft @jhendrixMSFT ping

**Comment by jhendrixMSFT** [comment]
Fix is in `azappconfig@0.4.2`
